### PR TITLE
feat: Implement BillingOptions for ApplePay

### PR DESCRIPTION
**Pull request by borisprimer**
Needs to be updated and retested after we release this on native SDK.

### CODE REVIEW

File: packages/sdk/ios/Sources/DataModels/PrimerSettings+Extensions.swift [hunk 1]
```diff
@@ -43,39 +43,65 @@ extension PrimerSettings {
...
+                            case "postalAddress":
+                                return .postalAddress
+                            default:
+                                return nil // Ignore unknown values or handle accordingly
+                            }
+                        }
+                    }
+
+                    billingOptions = PrimerApplePayOptions.BillingOptions(
+                        requiredBillingContactFields: requiredBillingContactFields
```
Comment (semirp) [inline packages/sdk/ios/Sources/DataModels/PrimerSettings+Extensions.swift:93]: [question]: do we need to filter `null` values?

File: packages/sdk/src/models/PrimerSettings.ts [hunk 2]
```diff
@@ -98,7 +98,6 @@ interface IPrimerApplePayOptions {
    * @deprecated Use Client Session API to provide merchant name value: https://primer.io/docs/payment-methods/apple-pay/direct-integration#prepare-the-client-session
    */
   merchantName?: string;
-  isCaptureBillingAddressEnabled?: boolean;
```
Comment (semirp) [inline packages/sdk/src/models/PrimerSettings.ts:101]: [question]: should we keep and deprecate this?

File: packages/sdk/ios/Sources/DataModels/PrimerSettings+Extensions.swift (hunk 1 above)
Comment (borisprimer) [reply packages/sdk/ios/Sources/DataModels/PrimerSettings+Extensions.swift:93]: We are already doing that by using `compactMap`, correct? @semirp

File: packages/sdk/src/models/PrimerSettings.ts (hunk 2 above)
Comment (borisprimer) [reply packages/sdk/src/models/PrimerSettings.ts:101]: Nice catch, it should be deprecated and live under `IPrimerApplePayOptions` not `IBillingOptions`. Thanks! Pushed the update

File: packages/sdk/src/models/PrimerSettings.ts [hunk 3]
```diff
@@ -111,13 +110,18 @@ interface IPrimerApplePayOptions {
...
 }
 interface IShippingOptions {
-  isCaptureShippingAddressEnabled: boolean;
-  additionalShippingContactFields?: AdditionalShippingContactField[];
+  shippingContactFields?: RequiredContactField[];
   requireShippingMethod: boolean;
 }
-type AdditionalShippingContactField = 'name' | 'emailAddress' | 'phoneNumber';
+interface IBillingOptions {
+  isCaptureBillingAddressEnabled: boolean;
```
[the remark this answers is not in the document]
Comment (borisprimer) [reply packages/sdk/src/models/PrimerSettings.ts]: exactly... stupid mistake. fixed.

File: packages/sdk/ios/Sources/DataModels/PrimerSettings+Extensions.swift (hunk 1 above)
Comment (semirp) [reply packages/sdk/ios/Sources/DataModels/PrimerSettings+Extensions.swift:93]: cool. didn't know about how it works

**Review by NQuinn27 — APPROVED** [review]

**Review by semirp — APPROVED** [review]

**Pull request merged**